Mage.Sets/src/mage/cards/r/RoryWilliams.java

```diff
@@ -0,0 +1,150 @@
+import mage.players.Player;
+
+
+/**
+ *
+ * @author Skiwkr
+ */
+public final class RoryWilliams extends CardImpl {
+
+    public RoryWilliams(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{W}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Partner with Amy Pond
+        this.addAbility(new PartnerWithAbility("Amy Pond"));
+
+        // First strike
+        this.addAbility(FirstStrikeAbility.getInstance());
+
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // The Last Centurion -- When you cast this spell from anywhere other than exile, exile it with three time counters on it. It gains suspend. Then investigate.
+        Ability ability = new RoryWilliamsTriggeredAbility(new RoryWilliamsEffect());
+        ability.addEffect(new InvestigateEffect(1).concatBy(", then"));
+        this.addAbility(ability);
+    }
+
+    private RoryWilliams(final RoryWilliams card) {
+        super(card);
+    }
+
+    @Override
+    public RoryWilliams copy() {
+        return new RoryWilliams(this);
+    }
+}
+
+class RoryWilliamsTriggeredAbility extends TriggeredAbilityImpl {
+
+    public static final String SOURCE_CAST_SPELL_ABILITY = "sourceCastSpellAbility";
+
+    public RoryWilliamsTriggeredAbility(Effect effect) {
+        this(effect, false);
+    }
+
+    public RoryWilliamsTriggeredAbility(Effect effect, boolean optional) {
+        super(Zone.STACK, effect, optional);
+        this.ruleAtTheTop = true;
+        setTriggerPhrase("When you cast this spell from anywhere other than exile, ");
+    }
+
+    protected RoryWilliamsTriggeredAbility(final RoryWilliamsTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public RoryWilliamsTriggeredAbility copy() {
+        return new RoryWilliamsTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.SPELL_CAST;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (!event.getSourceId().equals(this.getSourceId())) {
+            return false;
+        }
+        MageObject spellObject = game.getObject(sourceId);
+        if ((!(spellObject instanceof Spell))) {
+            return true;
+        }
+        Spell spell = (Spell) spellObject;
+        if (spell.getSpellAbility() != null) {
+            getEffects().setValue(SOURCE_CAST_SPELL_ABILITY, spell.getSpellAbility());
+        }
+        if (spell.getFromZone().toString().equals("exile zone")){
+            return false;
+        }
+        getEffects().setValue("spellCast", spell);
+        return true;
+    }
+}
+
+class RoryWilliamsEffect extends OneShotEffect {
```

Comment:
suggest using existing `ExileSpellWithTimeCountersEffect` and `GainSuspendEffect` rather than reimplementing that functionality

Reply:
GainSuspendEffect needs to be passed a MageObjectReference, am I able to generate that from inside the card body, and if so, what is the proper way to do so?  I would think I would just do new MageObjectReference(this, game), but I don't know how to get the game object, or if I should even be doing it that way.

Reply:
How are MageObjectReferences used?  I figure they're use to track cards through zones, but do I need to initialize it outside the card body or is there some card value I should be passing to get the correct MOR?

Either way, I wasn't able to get this to work in the last commit, let me know if you have any insight, otherwise I'll just reimplement the original hardcoded effect.

Reply:
Ah. Sorry, if `GainSuspendEffect` needs a `MageObjectReference` in constructor then it can only be used inside another effect. `game` parameter only accessible from within `apply()` method. That's frustrating, refactoring to clean that up is beyond the scope of this PR though.

You'll need to retain a custom `RoryWilliamsEffect`, but try to either inherit from `ExileSpellWithTimeCountersEffect` or call its apply method (whichever is cleaner) rather than duplicating its code.

Reply:
I added the gain suspend funcitonality to ExileSpellWithTimeCountersEffect so now you can just call the function and it'll do it all in one go. I was able to simplify [[Epochrasite]] with it this way as well